cachix/src/Cachix/Client/Push.hs

```diff
@@ -138,6 +140,18 @@ getCacheAuthToken :: PushSecret -> Token
 getCacheAuthToken (PushToken token) = token
 getCacheAuthToken (PushSigningKey token _) = token
 
+updateProgressBar :: MonadIO m => Maybe ProgressBar -> ConduitM ByteString ByteString m ()
+updateProgressBar Nothing = awaitForever Data.Conduit.yield
+updateProgressBar (Just pg) =
+  await
+    >>= maybe
```

Comment:
iiuc this is a `let m = await >>= maybe pass m`, and that means it's another `awaitForever`.